services/webhook/gitea.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package webhook
+
+import (
+	webhook_model "code.gitea.io/gitea/models/webhook"
+	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/secret"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/services/forms"
+)
+
+type (
+	// GiteaAuthHeaderMeta contains the authentication header metadata
+	GiteaAuthHeaderMeta struct {
+		Name     string                       `json:"name"`
+		Type     webhook_model.AuthHeaderType `json:"type"`
+		Username string                       `json:"username,omitempty"`
+		Password string                       `json:"password,omitempty"`
+		Token    string                       `json:"token,omitempty"`
+	}
+
+	// GiteaMeta contains the gitea webhook metadata
+	GiteaMeta struct {
+		AuthHeaderEnabled bool                `json:"auth_header_enabled"`
+		AuthHeaderData    string              `json:"auth_header,omitempty"`
+		AuthHeader        GiteaAuthHeaderMeta `json:"-"`
+	}
+)
+
+// GetGiteaHook returns decrypted gitea metadata
+func GetGiteaHook(w *webhook_model.Webhook, decryptFn secret.DecryptSecretCallable) *GiteaMeta {
+	s := &GiteaMeta{}
+
+	// Legacy webhook configuration has no stored metadata
+	if w.Meta == "" {
+		return s
+	}
```

Comment:
Would a new data migration be necessary instead of this code?

Reply:
Which data would be migrated? Is it already possible to add authorization tokens into webhooks?

Reply:
No, it's not possible at the moment. For Gitea hooks the whole meta field is still empty right now. A possible migration would be to add `{"auth_header_enabled":false}` as meta data to all of the already stored webhooks in the DB. Doing so would eliminate this if-statement. Currently, it prevents json decode errors.